cve_bin_tool/version_scanner.py

```diff
@@ -190,9 +191,14 @@ def scan_and_or_extract_file(self, ectx, filepath):
                 self.file_stack.pop()
 
     def recursive_scan(self, scan_path):
+
         with Extractor(logger=self.logger, error_mode=self.error_mode) as ectx:
             if os.path.isdir(scan_path):
                 for filepath in self.walker([scan_path]):
+                    for exclude_path in self.exclude_folders:
+                        if fnmatch.fnmatch(filepath, exclude_path + "*"):
+                            return
+
```

Comment:
You also don't need to do this, DirWalker will handle all this.

Reply:
> You also don't need to do this, DirWalker will handle all this.

I was trying that. But DirWalker wasn't excluding compressed files.

Reply:
![Screenshot from 2020-08-16 12-22-00](https://user-images.githubusercontent.com/35963992/90328592-1f95a280-dfbb-11ea-97db-6039b5f35d3a.png)
We are actually compressing the files in our directory directly using args["directory"] without using DirWalker

Reply:
![image](https://user-images.githubusercontent.com/35963992/90332112-d43dbd00-dfd7-11ea-8e80-6c9cda334e1d.png)
I will have to use this. Will that be okay?

Reply:
Inside DirWalk.walk():
  filenames[:] = [
                    i
                    for i in filenames
                    if self.pattern_match(i, self.pattern)
                    and not self.pattern_match_directory(i, self.file_exclude_pattern)
                ]

Reply:
I have tested it on my machine and it seems like there is some problem with pattern_match function.
Your solution (`text.startswith`) will work for simple pattern but won't work for patterns like: `*/temp/*.jpg`. 
So, we need to do it with fnmatch.

Reply:
Maybe we can use different functions as I used in the PR?
pattern_match for matching patterns. I read the **fnmatch** documentation. It seems like it only works while comparing strings or regular expressions. We can use it but we will have to append '*' to every string in the folders_to_exclude. Writing another function for this one case seems like a better option